drainer/syncer.go

```diff
@@ -418,23 +413,36 @@ ForLoop:
 			}
 
 			if s.filter.SkipSchemaAndTable(schema, table) {
-				log.Info("skip ddl", zap.String("schema", schema), zap.String("table", table),
+				log.Info("skip ddl by filter", zap.String("schema", schema), zap.String("table", table),
 					zap.String("sql", sql), zap.Int64("commit ts", commitTS))
-			} else if sql != "" {
-				s.addDDLCount()
-				beginTime := time.Now()
-				lastAddComitTS = binlog.GetCommitTs()
-
-				log.Info("add ddl item to syncer, you can add this commit ts to `ignore-txn-commit-ts` to skip this ddl if needed",
-					zap.String("sql", sql), zap.Int64("commit ts", binlog.CommitTs))
+				continue
+			}
 
-				err = s.dsyncer.Sync(&dsync.Item{Binlog: binlog, PrewriteValue: nil, Schema: schema, Table: table})
-				if err != nil {
-					err = errors.Annotatef(err, "add to dsyncer, commit ts %d", binlog.CommitTs)
-					break ForLoop
+			if !s.cfg.SyncDDL {
+				log.Info("skip ddl by SyncDDL setting to false", zap.String("schema", schema), zap.String("table", table),
+					zap.String("sql", sql), zap.Int64("commit ts", commitTS))
+				// A empty sql force it to evict the downstream table info.
+				if s.cfg.DestDBType == "tidb" || s.cfg.DestDBType == "mysql" {
+					binlog.DdlQuery = []byte("")
```

Comment:
how about adding a filed `ShouldSkiped` into `dsync.Item`, and ingnore these `dsync.Item` while we need to execute it?  code implementation may be simpler.

Reply:
also need some king like SHouldSkiped into pkg/Loader instead of empty string?

Reply:
https://github.com/pingcap/tidb-binlog/pull/890/commits/a17b05a2f99508e42c7038759398245ede7e59d4
Use explictly field to skip item @GregoryIan @suzaku PTAL